### PR TITLE
SQS ingester for alloy transmuter

### PR DESCRIPTION
**Pull request by iboss-ptk**
## What is the purpose of the change

To allow SQS to support alloy transmuter pool.

### CODE REVIEW

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:222]: nit: suggest to bubble up error instead. Either is caught and silently skipped anyways but errors are easier to debug and test.

Also, consider formatting a pool ID here

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:232]: Does this depend on the contract uploaded where the cw pool info is set?

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:225]: Would it be possible to encapsulate the new logic and/or the CW-pool-specific logic into a separate function and cover by unit tests?

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:477]: Is there a way we could unit test this?

Also, would appreciate a short method spec 🙏

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:493]: nit: let's please define types and JSON schemas at the top of the file for readability and ease of finding

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:512]: ditto

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:508]: I wonder what is the impact of this on block processing time given that we are adding a few cosmwasm pool queries.

Would it be possible to do a benchmark on the in-place testnet?
A) No alloyed pools
B) 20 alloyed pools

For both, measure samples over 10 or so blocks and take the average.

While cumbersome, I think this would instill confidence and significantly derisk the upcoming release

**Review by p0mvn — COMMENTED** [review]
Nice work!

Mostly nits, unit test request and a block process time benchmark request

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (iboss-ptk) [reply ingest/sqs/pools/transformer/pool_transformer.go:232]: exactly

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (iboss-ptk) [reply ingest/sqs/pools/transformer/pool_transformer.go:493]: I think leaving this here is better, since it's only being used within the function. That fact even more obvious after extracting these queries into small functions. and only return the content inside, not is wrapped structure.

File: ingest/sqs/pools/transformer/export_test.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/export_test.go:40]:
```suggestion
func (pi *poolTransformer) UpdateAlloyedTransmuterInfo(
```

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:468]: nit: could we move alloyed-transmute specific helpers into a separate file so that when orderbook logic is added, we don't keep bloating the same transformer file

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (iboss-ptk) [reply ingest/sqs/pools/transformer/pool_transformer.go]: My internet was a bit broken, couldn't download the snapshot, so I did it with clean state. Let me know if you still want mainnet state.

measure time running pool transformer `Transform` function

`no pool avg  = 0.0000813042s`
`10 pools avg = 0.0074192627s`

File: ingest/sqs/pools/transformer/alloy_transmuter.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/alloy_transmuter.go:61]: nit: consider defining the struct at the top of the file

File: ingest/sqs/pools/transformer/alloy_transmuter.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/alloy_transmuter.go:90]: Same for struct definition

File: ingest/sqs/pools/transformer/alloy_transmuter.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/alloy_transmuter.go:52]: Consider moving to constant at the top to enhance readability

File: ingest/sqs/pools/transformer/alloy_transmuter.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/alloy_transmuter.go:80]: ditto for constant at the top so that all these query definitions are all in one place

File: ingest/sqs/pools/transformer/alloy_transmuter_test.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/alloy_transmuter_test.go:28]: Why do we need to set protorev route for this test?

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:227]: What's this?

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:465]: can we format address here to help with debugging if future issues occur

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:463]: nit: consider defining query string at the top

File: ingest/sqs/service/writelistener/cosmwasmpool_write_listener.go
Comment (p0mvn) [inline ingest/sqs/service/writelistener/cosmwasmpool_write_listener.go:53]: The TODO seems to be addressed, is it not?

**Review by p0mvn — APPROVED** [review]
Nice work.

Approving with the understanding that the test on `v25.x` was done, including the ingest time measurement we discussed. Also, hoping that the remaining nits will be addressed async 🙏

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [reply ingest/sqs/pools/transformer/pool_transformer.go]: Looks reasonable to me. Thank you

File: ingest/sqs/pools/transformer/alloy_transmuter.go
Comment (p0mvn) [reply ingest/sqs/pools/transformer/alloy_transmuter.go:61]: I noticed that you already replied to this:
https://github.com/osmosis-labs/osmosis/pull/8291#discussion_r1628841399

Feel free to resolve any of the comments you disagree with. Let's not block on this

**Review by p0mvn — APPROVED** [review]

**Pull request merged**
